network_analysis/configure-subnetwork-trace/src/main/java/com/esri/samples/configure_subnetwork_trace/ConfigureSubnetworkTraceController.java

```diff
@@ -0,0 +1,419 @@
+
+          } else {
+            new Alert(Alert.AlertType.ERROR, "Trace result not a utility element.").show();
+          }
+        } catch (Exception e) {
+          new Alert(Alert.AlertType.ERROR,
+              "Error running utility network trace. For a working barrier condition, try \"Transformer Load\" Equal \"15\".")
+              .show();
+        }
+      });
+    } catch (Exception e) {
+      new Alert(Alert.AlertType.ERROR, "Error building trace parameters / configuration.").show();
+    }
+  }
+
+  /**
+   * Resets the trace configuration and UI back to the state at application start.
+   */
+  @FXML
+  private void onResetClick() {
+
+    // reset the utility trace configuration and traversability to the state at application start
+    utilityTraceConfiguration = initialUtilityTraceConfiguration;
+    utilityTraceConfiguration.getTraversability().setBarriers(initialExpression);
+
+    // show the configuration expression from the application start in the text area
+    traceConditionsTextArea.setText(expressionToString(initialExpression));
+
+    // un-check the checkboxes for including barriers and containers
+    includeContainersCheckBox.setSelected(false);
+    includeBarriersCheckBox.setSelected(false);
+
+    // select the first item in each ComboBox
+    comparisonSourcesComboBox.getSelectionModel().select(0);
+    comparisonOperatorsComboBox.getSelectionModel().select(0);
+  }
+
+  /**
+   * Updates the contents of the comparison value choices ComboBox depending on the selected comparison source.
+   */
+  @FXML
+  private void onComparisonSourceChanged() {
+
+    // clear any previous text input
+    comparisonValuesTextField.clear();
+
+    if (comparisonSourcesComboBox.getSelectionModel().getSelectedItem() != null) {
+
+      // determine if we need to show a selection of values in the combo box, or a text entry field
+      UtilityNetworkAttribute selectedAttribute = comparisonSourcesComboBox.getSelectionModel().getSelectedItem();
+      if (selectedAttribute.getDomain() instanceof CodedValueDomain) {
+
+        // populate and show the comparison values combo box
+        List<CodedValue> comparisonValues = ((CodedValueDomain) selectedAttribute.getDomain()).getCodedValues();
+        comparisonValuesComboBox.getItems().clear();
+        comparisonValuesComboBox.getItems().addAll(comparisonValues);
+        comparisonValuesComboBox.getSelectionModel().select(0);
+
+      } else {
+        comparisonValuesComboBox.getItems().clear();
+      }
+
+      // toggle the selection combo box to be visible if it has any items
+      comparisonValuesComboBox.setVisible(!comparisonValuesComboBox.getItems().isEmpty());
+      // toggle the text field to be hidden if the combo box is visible, or show it if the combo box is invisible
+      comparisonValuesTextField.setVisible(!comparisonValuesComboBox.isVisible());
+    }
+  }
+
+  /**
+   * Converts an object representing a value into the data type specified.
```

Comment:
I agree that this is quite confusing.
I've found that this particular utility network accepts `0` and `1` as a 'boolean', hence the text field only accepts numbers. 
I think this sample has a little more functionality than the dataset actually supports. I think the UN team designed it particularly this way so that users have a more general starting point rather than limiting it to the abilities of the demo dataset.
Maybe @jNery could also help clarify?